packages/react-sdk/docusaurus/docs/React/02-tutorials/02-audio-room.mdx

```diff
@@ -3,4 +3,543 @@ title: Audio Room Tutorial
+
+:::tip
+If you don't have a Stream account yet, you can register and try it for free [here](https://getstream.io/try-for-free/).
+:::
+
+The last setup step is to create a `.env` file. This is where we will put our API key to keep it safe.
+Create a `.env` file in the root of your project and paste this code in (and replace `<your_api_key>` with the real one):
+
+```
+VITE_STREAM_API_KEY=<your_api_key>
+```
+
+With that the basic setup of the project and the Stream SDK is ready and we can dive into the code.
+
+### User Authentication
+
+We won't cover the entire flow of authentication in this guide because we want to focus on the important parts.
+You can check the complete flow in the sample app repository, and we'll go through the important API calls now.
+
+The first important step is to initialize the Stream client. Therefore, we need the API key and a token provider.
+
+:::tip
+Want to learn why we need the token provider and how it works? See [this page](../../guides/client-auth/#token-providers) for more detail.
+:::
+
+First, we can collect the API key from our `.env` file by simply importing it like this:
+
+```
+const apiKey = import.meta.env.VITE_STREAM_API_KEY as string;
+```
+
+Next, to get a token, we need to create one in our backend. Whenever this expires the SDK will call the token provider again to renew the token.
+There are many different ways to solve this. In this example, we create (and - of course - memoize) a function that uses a `user` object in the same component.
+
+```tsx
+const tokenProvider = useCallback(async () => {
+  const { token } = await fetch(
+    '/api/auth/create-token?' +
+      new URLSearchParams({
+        api_key: apiKey,
+        user_id: user.id,
+      }),
+    {},
+  ).then((res) => res.json());
+  return token as string;
+}, [apiKey, user.id]);
+```
+
+Having this we can create a client object:
+
+```tsx
+const client = useCreateStreamVideoClient({
+  apiKey,
+  tokenOrProvider: tokenProvider,
+  user,
+});
+```
+
+Whenever a user hits the login button we can authenticate them using a call to `connectUser`:
+
+```tsx
+await client.connectUser(user, user.token);
+```
+
+For this to work we need a `User` object which is a simple typescript file with the following properties:
+
+```ts
+export interface User {
+  id: string;
+  name: string;
+  imageUrl: string;
+  token?: string;
+}
+```
+
+With this, the user will be logged in and authenticated versus the Stream backend.
+
+### Creating rooms via the API
+
+Now that we have user authentication out of our way we can start building up our app.
+The first thing we'll do is to allow users to create an audio room where others can then join.
+
+The data model for this needs a few properties. Create a new Typescript file, call it `AudioRoom` and fill it up with this:
+
+```ts
+export interface AudioRoom {
```

Comment:
I'd include the import statements in the code snippets, it's always useful to see which types coming from the SDK and which are user-defined